solver/types/calltrace.go

```diff
@@ -0,0 +1,50 @@
+package types
+
+// CallTrace is the result of debug_traceCall. Types kept to strings for simplicity.
+// TODO(kevin): move to lib/ethclient.
+type CallTrace struct {
+	From    string      `json:"from"`
+	To      string      `json:"to"`
+	Data    string      `json:"data"`
+	Value   string      `json:"value"`
+	Type    string      `json:"type"`
+	Gas     string      `json:"gas"`
+	GasUsed string      `json:"gasUsed"`
+	Input   string      `json:"input"`
+	Output  string      `json:"output"`
+	Error   string      `json:"error"`
+	Calls   []CallTrace `json:"calls,omitempty"`
+}
+
+// Map converts CallTrace to a map[string]any.
+func (t CallTrace) Map() map[string]any {
```

Comment:
Using this to not change return type of CheckRequest and check handlers. Was a bit easier, because we insert an {"error": "some error"} if something goes wrong with trace